956 data in cut plot distorted after ticking and unticking log scale in x axis

GuiMacielPereira opened the pull request:
**Description of work:**
- Fixed bug where data gets distorted from setting log scale to x axis and then setting it to linear again
- Restructured section of the code dealing with updating the log scales on the axes.

**To test:**
- Open cut plot with some example data
- Open plot options > tick Logarithm for x axis > 'OK'
- Check log scale is applied to x axis
- Open plot options again > untick Logarithm for x axis > 'OK'
- Check plot is back to original format
- Repeat the same steps for y axis
- Now put both axis on a linear scale
- Go to Information > Bragg Peaks > check Aluminium
- Change y axis to log scale
- The lines for the Aluminium bragg peaks should stay symmetrical around zero
- Change y axis to linear
- The lines for the Aluminium bragg peaks should stay symmetrical around zero


Fixes #956

GuiMacielPereira commented on src/mslice/plotting/plot_window/cut_plot.py:
```diff
@@ -676,21 +638,58 @@ def x_log(self):
+        if value:
+            ydata = [line.get_ydata() for line in current_axis.get_lines()]
+            y_axis_min = get_min(ydata, absolute_minimum=0.)
+            linthresh_val = pow(10, np.floor(np.log10(y_axis_min)))
```
Add to tweak things a little bit, and ended up replacing a function called `get_min`. I am confident the functionality is the exact same, but my implementation uses numpy more effectively

robertapplin commented:
Thanks for the changes. Just did some functional testing and I get an error specifically for this part of the instructions:

Go to Information > Bragg Peaks > check Aluminium
Change y axis to log scale

```
Traceback (most recent call last):
  File "C:\Users\mlc47243\Documents\mantid-dev\mslice\src\mslice\presenters\plot_options_presenter.py", line 126, in get_new_config
    setattr(self._model, key, value)
  File "C:\Users\mlc47243\Documents\mantid-dev\mslice\src\mslice\plotting\plot_window\cut_plot.py", line 675, in y_log
    linthresh_val = _gca_sym_log_linear_threshold(axis_data)
  File "C:\Users\mlc47243\Documents\mantid-dev\mslice\src\mslice\plotting\plot_window\cut_plot.py", line 36, in _gca_sym_log_linear_threshold
    axis_data = np.array(axis_data)
ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 1 dimensions. The detected shape was (2,) + inhomogeneous part.
```

GuiMacielPereira commented:
My goodness, well spotted. I'll fix that.

GuiMacielPereira commented:
- Fixed it. This happened because I didn't consider the case where converting a list of arrays with different dimensions into a single array fails when using np.array(). Replaced it by np.concatenate(), as it's not important to preserve the dimensions.

robertapplin reviewed (approved):
Looks great - the bug is fixed and the code is much improved to before
